src/app.py

```diff
@@ -84,29 +98,36 @@ def process_workflow_job():
             ).seconds
             # delete from memory
             del jobs[job_id]
-        msg = (
-            f"action={action} repository={repository} job_id={job_id}"
-            f' workflow="{workflow}" time_to_finish={time_to_finish}'
-        )
+
+        extra_data = {
+            "time_to_finish": time_to_finish,
+            "conclusion": conclusion
+        }
+        context_details = {**context_details, **extra_data}
     else:
         app.logger.warning(f"Unknown action {action}, removing from memory")
         if job_id in jobs:
             del jobs[job_id]
-        msg = None
+        context_details = None
 
-    if msg:
-        app.logger.info(msg)
+    if context_details:
+        app.logger.info(dict_to_logfmt(context_details))
     return True
 
 
+allowed_events = {
+    "workflow_job": process_workflow_job
+}
+
+
 @app.route("/github-webhook", methods=["POST"])
 def github_webhook_process():
     event = request.headers.get(GithubHeaders.EVENT.value)
-    command = f"process_{event}"
 
-    if command == "process_workflow_job":
-        app.logger.debug(f"Calling function {command}")
-        process_workflow_job()
+    if event in allowed_events:
+        app.logger.debug(f"Calling function to process {event=}")
+        func = allowed_events.get(event)
```

Comment:
this is not a big deal but we already know `event` is in `allowed_events`, I would use `[]` instead of `.get()`

Reply:
also I would call it directly `allowed_events[event]()`

Reply:
That will work for sure, but I'm forcing code readability in here.

Reply:
fair enough